misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java

```diff
@@ -176,6 +177,15 @@ protected JsonElement transformSyntaxNode(Node node) {
                 // Ignore as semantic API calls cannot break the ST JSON creation.
             }
 
+            if (node.kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION) {
+                RemoteMethodCallActionNode remoteMethodCallActionNode = (RemoteMethodCallActionNode) node;
+                Optional<Symbol> expressionSymbol = this.semanticModel.symbol(remoteMethodCallActionNode.expression());
+                if (expressionSymbol.isPresent()) {
```

Comment:
Please check for the symbol kind before casting, for safety.

Reply:
fixed